domain/model/state/modelstate.go

```diff
@@ -219,6 +219,34 @@ func (s *ModelState) GetModelMetrics(ctx context.Context) (coremodel.ModelMetric
 	}, nil
 }
 
+// GetModelCloudType returns the cloud type from a read-only model that has been
+// set in the database. If no model has been set then an error satisfying
+// [modelerrors.NotFound] is returned.
+func (s *ModelState) GetModelCloudType(ctx context.Context) (string, error) {
+	db, err := s.DB()
+	if err != nil {
+		return "", errors.Capture(err)
+	}
+
+	m := dbReadOnlyModel{}
+	stmt, err := s.Prepare(`SELECT &dbReadOnlyModel.cloud_type FROM model`, m)
+	if err != nil {
+		return "", errors.Capture(err)
+	}
+
+	if err := db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
```

Comment:
nit: Can we remove the if statement wrap around this closure? Just makes it harder to read and grok what is going on.